src/main/java/edu/utdallas/davisbase/compiler/Compiler.java

```diff
@@ -689,4 +728,87 @@ private void validateNotCatalogTable(String tableName) throws CompileException {
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.SMALLINT && castNonNull(table.readSmallInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.INT && castNonNull(table.readInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.BIGINT && castNonNull(table.readBigInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.FLOAT && castNonNull(table.readFloat(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.DOUBLE && castNonNull(table.readDouble(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.YEAR && castNonNull(table.readYear(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.TIME && castNonNull(table.readTime(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.DATETIME && castNonNull(table.readDateTime(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.DATE && castNonNull(table.readDate(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.TEXT && castNonNull(table.readText(colIndex)).equalsIgnoreCase(value.toString())){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else{
+          return;
+        }
+      }
+    }
+    else{
+      return;
+    }
+  }
+
+  /**
+   * Checks DAVISBASE_COLUMNS table to see if given column has UNIQUE constraint
+   * @param tableName table name to check
+   * @param columnName column name within given table
+   * @return whether given column has UNIQUE constraint
+   * @throws IOException
+   * @throws StorageException
+   */
+  public boolean isUnique(String tableName, String columnName)throws IOException, StorageException {
+    TableFile table = context.openTableFile(CatalogTable.DAVISBASE_COLUMNS.getName());
+    while (table.goToNextRow()) {
+      if (castNonNull(table.readText(DavisBaseColumnsTableColumn.COLUMN_NAME.getOrdinalPosition()))
+        .equalsIgnoreCase(columnName)
+        && castNonNull(
+        table.readText(DavisBaseColumnsTableColumn.TABLE_NAME.getOrdinalPosition()))
+        .equalsIgnoreCase(tableName)) {
+        return BooleanUtils.fromText(castNonNull(
+          table.readText(DavisBaseColumnsTableColumn.IS_UNIQUE.getOrdinalPosition())));
+      }
+    }
+    return false;
```

Comment:
Would it be more appropriate to throw an `IllegalStateException` if we ever reach here?

As in, we're assuming that the `tableName` and `columnName` arguments actually exist in the catalog, so if we don't find them, then that's an application state issue. Alternatively, if this method is _intended_ to operate even when the combination of `tableName` and `columnName` does not already exist in the catalog, then please either rename the method to something more indicative of that fact and/or explicitly make a note of it in this method's JavaDoc.

Reply:
updated to throw an IllegalStateException